src/Reflection/Mutation/AddClassMethod.php

```diff
@@ -0,0 +1,29 @@
+<?php
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\Reflection\Mutation;
+
+use PhpParser\Node\Stmt\ClassMethod;
+use Roave\BetterReflection\Reflection\Mutator\ReflectionClassMutator;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+
+class AddClassMethod
+{
+    /**
+     * @var ReflectionClassMutator
+     */
+    private $mutator;
+
+    public function __construct()
+    {
+        $this->mutator = new ReflectionClassMutator();
+    }
+
+    public function __invoke(ReflectionClass $classReflection, string $methodName) : ReflectionClass
```

Comment:
No visibility? No parameters? What's going on here?

Reply:
I implemented the same version: https://github.com/Roave/BetterReflection/blob/master/src/Reflection/ReflectionClass.php#L1407

I think the idea was that you should you other methods (now mutations) to add parameters, body etc.

However it would be good to add modifiers here at least.